### ISSUE TITLE
Bug - Mention with @ is wrong

### ISSUE THREAD

**Issue by BurtGummer** [znuny/Znuny#532]
Hello,

that's the only reason why we can't use this great feature: We use a number as login, not the real name. This means that @12345 and not @firstname lastname is displayed in the ticket body. I can search with the Realname and I got a list with "Firstname Lastname". But after I select a User, its changed to @12345.

Is there any way to configure WHAT is displayed at this point?

Znuny 6.5.5, Debian 11, OpenLDAP.

**Comment by tgurr** [comment]
I initially tried to fix this but failed with https://github.com/znuny/Znuny/pull/436 (and failed because sadly its not that simple as I initially thought) and was adviced to create an issue in the feature-request repository instead https://github.com/znuny/znuny-feature-requests/issues/51. So effectivly this is a "duplicate" however I also consider this rather a bug than a feature and sadly there hasn't been any activity in trying to address this issue yet as we're in exactly the same boat as @BurtGummer.

**Comment by BurtGummer** [comment]
there is also a thread about in the forum: https://community.znuny.org/viewtopic.php?f=35&t=43841&p=177871

**Comment by mo-azfar** [comment]
Mention fullname has been put as roadmap for 7.x here.. https://www.znuny.org/en/roadmap

Quick work around:

1. Admin > Mentions::RichTextEditor###Templates
- Users > OutputTemplate > 
`{fullname} (<a class="Mention" href="#" title="{fullname}">@{username}</a>)`

2. https://github.com/znuny/Znuny/blob/dc3ab69eda71887841fc65602e6321002fe21167/Kernel/Output/HTML/TicketZoom/Mentions.pm#L54

```
UserEmail => $User{UserEmail},
(+) UserFullname => $User{UserFullname},
```


3. https://github.com/znuny/Znuny/blob/dc3ab69eda71887841fc65602e6321002fe21167/Kernel/Output/HTML/Templates/Standard/AgentTicketZoom/MentionsTable.tt#L20

```
(-) <span>[% Data.UserLogin | html %]</span>
(+) <span>[% Data.UserFullname | html %] - [% Data.UserLogin | html %]</span>
```

Mention text
![mention](https://github.com/znuny/Znuny/assets/62957738/bafbf34b-2689-4ab9-a51f-abbf94b3c5cb)

Mention dashboard
![mention-dashboard](https://github.com/znuny/Znuny/assets/62957738/a8c6c783-5fd2-4ba9-ac6f-2cac54ab9a00)

Mention history
![mention-history](https://github.com/znuny/Znuny/assets/62957738/f1ff17e4-2f7a-4ee4-a2c2-4711a508d6a9)

Mention toolbar
![mention-toolbar](https://github.com/znuny/Znuny/assets/62957738/0af3fadb-aacd-4845-ad9d-59b0b2514313)

Mention ticket widget
![mention-widget](https://github.com/znuny/Znuny/assets/62957738/29905e63-2eaf-4238-a055-253462913e33)


**I DID NOT CHECK AGAINST EMAIL. DO TEST YOURSELF**

**Comment by hanneshal** [comment]
Hi, 

you both are correct. It is not that simple and it would be nice to have it changed.

Maybe it helps @tgurr when I explain how this all works together:

A user creates a mention in the gui.
We store this info in a link, which is later identified by this code:

>Kernel::System::Ticket::Event::Mentions;
>L:70
```Perl

    my $MentionedUserIDs = $MentionObject->GetMentionedUserIDsFromString(
        HTMLString      => $HTMLBody,
        PlainTextString => $Article{Body} // '',
        Limit           => $MentionsLimit,
    );

```
In 
> Kernel::System::Mention;
> L:559

the check against that is performed. 

```Perl
sub GetMentionedUserIDsFromString {
    ...
    ...
    my @MentionedUsers = (
        $Param{HTMLString}
            =~ m{<a\b[^>]*?\bclass="Mention"[^>]*?>\Q$MentionsTriggerConfig->{User}\E(.*?)<\/a>}sg
    );

    my @MentionedGroups = (
        $Param{HTMLString}
            =~ m{<a\b[^>]*?\bclass="GroupMention"[^>]*?>\Q$MentionsTriggerConfig->{Group}\E(.*?)<\/a>}sg
    );
```

There are multiple ways to change this: 
- Move the id into a data tag, within the href and diplay a different string (firstname/lastname)
- Just replace the logic for insert and check

and of course, replace the output template. 

```xml
    <Setting Name="Mentions::RichTextEditor###Templates" Required="0" Valid="1">
        <Description Translatable="1">Frontend registration of input/output templates for mention plugin of CKEditor.</Description>
        <Navigation>Core</Navigation>
        <Value>
            <Hash>
                <Item Key="Users">
                    <Hash>
                        <Item Key="ItemTemplate">&lt;li class="Mention" data-id="{id}"&gt;{fullname}&lt;/li&gt;</Item>
                        <Item Key="OutputTemplate">&lt;a class="Mention" href="#"&gt;@{username}&lt;/a&gt; </Item>
                    </Hash>
                </Item>
                <Item Key="Groups">
                    <Hash>
                        <Item Key="ItemTemplate">&lt;li class="Mention" data-id="{id}"&gt;{name} ({mentionLabel})&lt;/li&gt;</Item>
                        <Item Key="OutputTemplate">&lt;a class="GroupMention" href="#"&gt;{name}&lt;/a&gt; </Item>
                    </Hash>
                </Item>
            </Hash>
        </Value>
    </Setting>
```

maybe this points to the correct direction to update the PR. As already mentioned, it is on the roadmap for 7.1. 
A backport is not planned at the moment. 

Regards and thank you for contributing. 
Johannes

**Comment by BurtGummer** [comment]
> Mention fullname has been put as roadmap for 7.x here.. https://www.znuny.org/en/roadmap
> 
> Quick work around:
> 
> 1. Admin > Mentions::RichTextEditor###Templates
> 
> * Users > OutputTemplate >
>   `{fullname} (<a class="Mention" href="#" title="{fullname}">@{username}</a>)`
[...]

Does someone try this solution? Normaly I dont like changing the code .. 🤔

**Comment by NiklasSchmitt** [comment]
The feature to use the user fullname as mention label will be fixed within the upcoming Znuny 7.1.1
